lib/collection/src/operations/verification/mod.rs

```diff
@@ -0,0 +1,177 @@
+mod search;
+
+use std::fmt::Display;
+
+use segment::types::Filter;
+use tokio::sync::RwLockReadGuard;
+
+use super::config_diff::StrictModeConfigDiff;
+use crate::collection::Collection;
+
+// Creates a new `VerificationPass` for successful verifications.
+// Don't use this, unless you know what you're doing!
+pub fn new_pass() -> VerificationPass {
+    VerificationPass { inner: () }
+}
+
+/// A pass, created on successful verification.
+pub struct VerificationPass {
```

Comment:
Let's rollback verification pass for now, and keep config and `StrictModeVerification` trait with implementation.

Changing Access logic might be non-trivial enough to make a separate discussion about it